lib/web_ui/skwasm/surface.cpp

```diff
@@ -77,13 +94,23 @@ class Surface {
     emscripten_dispatch_to_thread(
         _thread, EM_FUNC_SIG_VII,
         reinterpret_cast<void*>(fNotifyRenderComplete), nullptr, this,
-        renderId);
-    return renderId;
+        callbackId);
+    return callbackId;
+  }
+
+  uint32_t rasterizeImage(SkImage* image, ImageByteFormat format) {
+    uint32_t callbackId = ++_currentCallbackId;
+    image->ref();
+
+    emscripten_dispatch_to_thread(_thread, EM_FUNC_SIG_VIIII,
+                                  reinterpret_cast<void*>(fRasterizeImage),
+                                  nullptr, this, image, format, callbackId);
+    return callbackId;
   }
 
   // Main thread only
```

Comment:
Is there a way to validate this property at runtime?

Reply:
Yes, I think I found one. I can make it an assert so that it only executes in debug mode.